mapnik_c_api.cpp

```diff
@@ -255,6 +269,15 @@ mapnik_image_blob_t * mapnik_image_to_png_blob(mapnik_image_t * i) {
     return blob;
 }
 
+const char * mapnik_version_string() {
+#if MAPNIK_VERSION >= 200100
+	return MAPNIK_VERSION_STRING;
+#else
+#define MAPNIK_C_API_STRINGIFY(n) #n
+	return "ABI " MAPNIK_C_API_STRINGIFY(MAPNIK_VERSION);
+#endif
+}
```

Comment:
Versions prior to 2.1 do not have the MAPNIK_VERSION_STRING macro, so I fall back to, e.g., 200000. If I did not, Travis (which uses 2.0) can at least compile. With newer versions the whole #else block is never even used.

On 10. Juli 2015 22:07:28 MESZ, Dane Springmeyer <notifications@github.com> wrote:
>> @@ -255,6 +269,15 @@ mapnik_image_blob_t *
>mapnik_image_to_png_blob(mapnik_image_t * i) {
>>      return blob;
>>  }
>>  
>> +const char * mapnik_version_string() {
>> +#if MAPNIK_VERSION >= 200100
>> +	return MAPNIK_VERSION_STRING;
>> +#else
>> +#define MAPNIK_C_API_STRINGIFY(n) #n
>> +	return "ABI " MAPNIK_C_API_STRINGIFY(MAPNIK_VERSION);
>> +#endif
>> +}
>
>This looks really odd. What are you trying to accomplish? Why put an
>`ABI ` in front when using more recent mapnik? Also `mapnik/config.hpp`
>does not include any C++ so it can be safely included by C apps.
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/springmeyer/mapnik-c-api/pull/10/files#r34391364

-- 
Von meinem Mobiltelefon gesendet.